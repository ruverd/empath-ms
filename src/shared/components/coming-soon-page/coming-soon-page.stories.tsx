import type { Meta, StoryObj } from "@storybook/react";
import { ComingSoonPage } from "./coming-soon-page";

const meta = {
  title: "Components/ComingSoonPage",
  component: ComingSoonPage,
  parameters: {
    layout: "fullscreen",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof ComingSoonPage>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    featureName: "Analytics Dashboard",
  },
};

export const LongFeatureName: Story = {
  args: {
    featureName:
      "Advanced Machine Learning and AI-Powered Recommendations System",
  },
};

export const ShortFeatureName: Story = {
  args: {
    featureName: "Reports",
  },
};
