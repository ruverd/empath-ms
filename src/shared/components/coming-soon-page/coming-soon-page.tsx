import { Badge } from "../ui/badge";

interface ComingSoonPageProps {
  featureName: string;
}

export const ComingSoonPage = ({ featureName }: ComingSoonPageProps) => {
  return (
    <div className="flex min-h-[calc(100vh-68px)] items-center justify-center p-6">
      <div className="relative flex flex-col items-center justify-center p-8 border border-primary-violet rounded-2xl bg-gradient-to-b from-[rgba(17,0,55,0.7)] to-[#2B1553] max-w-lg w-full gap-6">
        <Badge variant="default" className="absolute -top-3">
          Coming Soon
        </Badge>
        <div className="flex flex-col items-center text-center gap-4">
          <h1 className="text-2xl font-bold text-shades-violet-8">
            {featureName}
          </h1>
          <p className="text-base text-secondary-foreground">
            We're working hard to bring you an amazing{" "}
            {featureName.toLowerCase()} experience. Stay tuned for exciting
            updates as we enhance our platform to serve you better!
          </p>
        </div>
      </div>
    </div>
  );
};
