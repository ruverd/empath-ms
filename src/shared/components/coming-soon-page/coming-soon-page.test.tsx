import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { ComingSoonPage } from "./coming-soon-page";

describe("ComingSoonPage", () => {
  it("should renders with feature name", () => {
    render(<ComingSoonPage featureName="Analytics" />);

    expect(screen.getByText("Analytics")).toBeInTheDocument();
    expect(screen.getByText("Coming Soon")).toBeInTheDocument();
  });

  it("should renders description with feature name in lowercase", () => {
    render(<ComingSoonPage featureName="Analytics" />);

    expect(
      screen.getByText(
        /we're working hard to bring you an amazing analytics experience/i
      )
    ).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<ComingSoonPage featureName="Analytics" />);
    expect(container).toMatchSnapshot();
  });
});
