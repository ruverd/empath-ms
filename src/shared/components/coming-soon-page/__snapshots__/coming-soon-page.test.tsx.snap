// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ComingSoonPage > should match snapshot 1`] = `
<div>
  <div
    class="flex min-h-[calc(100vh-68px)] items-center justify-center p-6"
  >
    <div
      class="relative flex flex-col items-center justify-center p-8 border border-primary-violet rounded-2xl bg-gradient-to-b from-[rgba(17,0,55,0.7)] to-[#2B1553] max-w-lg w-full gap-6"
    >
      <div
        class="inline-flex items-center border-shades-violet-5 rounded-full border px-2.5 py-1 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-primary-cloud bg-shades-violet-3 absolute -top-3"
      >
        Coming Soon
      </div>
      <div
        class="flex flex-col items-center text-center gap-4"
      >
        <h1
          class="text-2xl font-bold text-shades-violet-8"
        >
          Analytics
        </h1>
        <p
          class="text-base text-primary-cloud"
        >
          We're working hard to bring you an amazing
           
          analytics
           experience. Stay tuned for exciting updates as we enhance our platform to serve you better!
        </p>
      </div>
    </div>
  </div>
</div>
`;
